src/lib/apiTypesV2/Subscription.h

```diff
@@ -0,0 +1,78 @@
+#ifndef SRC_LIB_APITYPESV2_SUBSCRIPTION_H
+#define SRC_LIB_APITYPESV2_SUBSCRIPTION_H
+
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+
+#include <string>
+#include <vector>
+
+#include "../ngsi/Duration.h"
+#include "../ngsi/Throttling.h"
+
+struct EntID
+{
+  std::string id;
+  std::string idPattern;
+  std::string type;
+  std::string toJson();
+};
```

Comment:
Strange for a man worried about cycles and assembler instructions, to have a unecessary vector going back and forth in a lll assignments, copies, ... Just for efficiency, the calls should have only the necessary member. If we have a vector of 10000 entititesID we should carry with 10000 vectors of attributes?
I see two concepts, an entity and an entityID. Sometimes a concept is representable in a basic type (perfect as a member) and sometimes it requieres a more elaborated representation -> a class. In this case , it has id, idPattern and type
Maybe the Entity class in v2 should be refactored to use EntID.
Using entity ids without the need  of the whole entity seems reasonable and convenient